raft.go

```diff
@@ -31,6 +31,8 @@ var (
 func (r *Raft) getRPCHeader() RPCHeader {
 	return RPCHeader{
 		ProtocolVersion: r.config().ProtocolVersion,
+		ID:              []byte(r.config().LocalID),
+		Addr:            r.trans.EncodePeer(r.config().LocalID, r.localAddr),
```

Comment:
I'm pretty sure adding these new fields to the `RPCHeader` is backwards compatible, but in case we wanted to verify it, I think something like this works as an "upgrade test" without having to spin up a bunch of processes:

1. create a copy of the old struct definitions, maybe even just in a test package called `AppendEntriesRequestV3` and `RPCHeaderV3`
2. msgpack encode something with the new struct, and show it can be decoded with the old V3 structs
3. msgpack encode something with the V3 structs, and show it can be decoded correctly with the new structs

I did this once in a private repo and it seemed fine, but I thought I'd mention in case we want to be extra sure that it works.

Reply:
After discussion with @mkeeler this test is not really needed as we know it's safe to add parameters to msgpack